src/marker-parser.js

```diff
@@ -328,13 +337,16 @@ export default class MarkerParser {
                     lineNumber,
                 );
             } else {
+                // Turns out that an empty optional tag group, though typed as
+                // string can actually be undefined. So, for the optional bits
+                // we'll just make sure they get coerced to empty strings.
                 this._recordMarkerStart(
-                    startDecode[1],
-                    startDecode[3],
+                    startDecode[TagDecodeGroup.tagId],
+                    startDecode[TagDecodeGroup.targetFileName],
                     lineNumber,
-                    startDecode[2],
+                    startDecode[TagDecodeGroup.checksum] || "",
                     startMatch[1],
-                    startDecode[4],
+                    startDecode[TagDecodeGroup.commentEnd] || "",
```

Comment:
Nice way to document which group contains what.

Reply:
Thanks. I was getting confused and thought some self-documenting code might help :)